packages/@aws-cdk/cloudformation-diff/lib/diff/template-and-changeset-diff-merger.ts

```diff
@@ -0,0 +1,156 @@
+
+  public static determineChangeSetReplacementMode(propertyChange: ChangeSetResourceChangeDetail): types.ReplacementModes {
+    if (propertyChange.Target?.RequiresRecreation === undefined) {
+      // We can't determine if the resource will be replaced or not. That's what conditionally means.
+      return 'Conditionally';
+    }
+
+    if (propertyChange.Target.RequiresRecreation === 'Always') {
+      switch (propertyChange.Evaluation) {
+        case 'Static':
+          return 'Always';
+        case 'Dynamic':
+          // If Evaluation is 'Dynamic', then this may cause replacement, or it may not.
+          // see 'Replacement': https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_ResourceChange.html
+          return 'Conditionally';
+      }
+    }
+
+    return propertyChange.Target.RequiresRecreation as types.ReplacementModes;
+  }
+
+  changeSet: DescribeChangeSetOutput | undefined;
+  changeSetResources: types.ChangeSetResources;
+
+  constructor(
+    args: {
+      changeSet: DescribeChangeSetOutput;
+      changeSetResources?: types.ChangeSetResources; // used for testing -- otherwise, don't populate
+    },
+  ) {
+    this.changeSet = args.changeSet;
+    this.changeSetResources = args.changeSetResources ?? this.convertDescribeChangeSetOutputToChangeSetResources(this.changeSet);
+  }
+
+  /**
+   * Read resources from the changeSet, extracting information into ChangeSetResources.
+   */
+  private convertDescribeChangeSetOutputToChangeSetResources(changeSet: DescribeChangeSetOutput): types.ChangeSetResources {
+    const changeSetResources: types.ChangeSetResources = {};
+    for (const resourceChange of changeSet.Changes ?? []) {
+      if (resourceChange.ResourceChange?.LogicalResourceId === undefined) {
+        continue; // Being defensive, here.
+      }
+
+      const propertyReplacementModes: types.PropertyReplacementModeMap = {};
+      for (const propertyChange of resourceChange.ResourceChange.Details ?? []) { // Details is only included if resourceChange.Action === 'Modify'
+        if (propertyChange.Target?.Attribute === 'Properties' && propertyChange.Target.Name) {
+          propertyReplacementModes[propertyChange.Target.Name] = {
+            replacementMode: TemplateAndChangeSetDiffMerger.determineChangeSetReplacementMode(propertyChange),
+          };
+        }
+      }
+
+      changeSetResources[resourceChange.ResourceChange.LogicalResourceId] = {
+        resourceWasReplaced: resourceChange.ResourceChange.Replacement === 'True',
+        resourceType: resourceChange.ResourceChange.ResourceType ?? TemplateAndChangeSetDiffMerger.UNKNOWN_RESOURCE_TYPE, // DescribeChangeSet doesn't promise to have the ResourceType...
+        propertyReplacementModes: propertyReplacementModes,
+        beforeContext: _maybeJsonParse(resourceChange.ResourceChange.BeforeContext),
+        afterContext: _maybeJsonParse(resourceChange.ResourceChange.AfterContext),
```

Comment:
In my next change, I will either remove these or use them... depending on if I can get DescribeChangeSet with IncludePropertyValues to cooperate

Reply:
I decided to remove them